src/nodeProvider.ts

```diff
@@ -222,56 +241,96 @@ export class DidactNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 		return children;
 	}
 
-	findCategoryNode(category : string) : TreeNode | undefined {
+	findCategoryNode(category : string) : SimpleNode | null {
 		const nodeToFind = new TreeNode(category, category, undefined, vscode.TreeItemCollapsibleState.Collapsed);
-		if (this.doesNodeExist(this.treeNodes, nodeToFind)) {
-			return nodeToFind;
-		}
-		return undefined;
+		const foundNode = this.getCategory(this.treeNodes, nodeToFind);
+		return foundNode;
 	}
 	
-	async findTutorialNode(category : string, tutorialName : string ) : Promise<TreeNode | undefined> {
+	async findTutorialNode(category : string, tutorialName : string ) : Promise<TutorialNode | undefined> {
 		const catNode = this.findCategoryNode(category);
 		if (catNode) {
-			const treeItems : TreeNode[] = await this.getChildren(catNode);
+			const treeItems : SimpleNode[] = await this.getChildren(catNode);
 			let foundNode : TreeNode | undefined = undefined;
 			treeItems.forEach(element => {
-				if (element.label === tutorialName && element.category === category) {
-					foundNode = element;
+				if (element instanceof TreeNode) {
+					const elementAsTreeNode = element as TreeNode;
```

Comment:
Sonarlint is reporting that "This assertion is unnecessary since it does not change the type of the expression.sonarlint(typescript:S4325)
"